---
title: "Project report"
author: "Mario Lim√≥n"
date: "2025-02-02"
output: 
  pdf_document:
    toc: yes
    toc_depth: 3
bibliography: "./Proyect_references.bib"
---
## Data process

#### Libraries
```{r}
library("recount3")
library("ggplot2")
library("edgeR")
library("limma")
```

#### Download data
Downloading data from recount 3 database of project 'Transcriptome profiling in
knock-in mouse models of Huntington''s disease  [Striatum_mRNA] (SRP053398). The
data of this study is to analyze the expression profiles of RNA in mouse models 
with Hungtington's diseaase at different ages with different lengths of CAG 
repeats, to understand better the molecular changes related with this neurodegenarative 
disease.


```{r}
rse_gene_SRP053398 <- recount3::create_rse_manual(
  project = "SRP053398",
  project_home = "data_sources/sra",
  organism = "mouse",
  annotation = "gencode_v23",
  type = "gene")
```
#### Processing data

```{r}
# Transform the counts
assay(rse_gene_SRP053398, "counts") <- compute_read_counts(rse_gene_SRP053398)

# Before expanding the data we should explore the attributes
rse_gene_SRP053398$sra.sample_attributes[1:3]
```

```{r}
# Now we expand the attributes to process the data to make easier the analysis
rse_gene_SRP053398 <- expand_sra_attributes(rse_gene_SRP053398)
colData(rse_gene_SRP053398)[
    ,
    grepl("^sra_attribute", colnames(colData(rse_gene_SRP053398)))]
```


```{r}
# To perform the statistical analysis we need to use a correct R format
rse_gene_SRP053398$sra_attribute.age <- as.numeric(gsub(" month", "", rse_gene_SRP053398$sra_attribute.age))
rse_gene_SRP053398$sra_attribute.genotype <- factor(tolower(rse_gene_SRP053398$sra_attribute.genotype))
rse_gene_SRP053398$sra_attribute.Sex <- factor(rse_gene_SRP053398$sra_attribute.Sex)

# We can check the summary of the interest variables
summary(as.data.frame(colData(rse_gene_SRP053398)[
    ,
    grepl("^sra_attribute.[age|disease|RIN|sex]", colnames(colData(rse_gene_SRP053398)))
]))
```
We will check the quality of the data calculating the proportion of assigned genes.
```{r}
rse_gene_SRP053398$assigned_gene_prop <- rse_gene_SRP053398$recount_qc.gene_fc_count_all.assigned / 
  rse_gene_SRP053398$recount_qc.gene_fc_count_all.total
summary(rse_gene_SRP053398$assigned_gene_prop)
```
```{r}
# We can create another variable to make more general the analysis
rse_gene_SRP053398$state <- factor(ifelse(rse_gene_SRP053398$sra_attribute.genotype != 'wild type' , "knock-in", "wildtype"))
table(rse_gene_SRP053398$state)
```


```{r}
hist(rse_gene_SRP053398$assigned_gene_prop, main = "Assigned genes", xlab = "Proportion of assigned genes")
```
```{r}
# We can also check if there is difference in the quality of the data between the different states
with(colData(rse_gene_SRP053398), tapply(assigned_gene_prop, state, summary))
```
#### Normalization of the data

Normalizing data is essential because it ensures that all features contribute 
equally to the analysis, preventing bias towards variables with larger scales.
```{r}
# With the library of edgeR we will normalize the data
dge <- DGEList(
    counts = assay(rse_gene_SRP053398, "counts"),
    genes = rowData(rse_gene_SRP053398)
)
dge <- calcNormFactors(dge)
```
## Expression data analyisis

#### Data distribution plots

```{r}
ggplot(as.data.frame(colData(rse_gene_SRP053398)), aes(y = assigned_gene_prop, x = state,)) +
    geom_violin(trim = FALSE, alpha = 0.3, fill = 'red') +
    geom_boxplot(width = 0.15) +
    theme_bw(base_size = 20) +
    ylab("Assigned Gene Prop") +
    xlab("Age Group")
```
#### Statistical model
```{r}
mod <- model.matrix(~ state + sra_attribute.Sex + sra_attribute.age + assigned_gene_prop,
    data = colData(rse_gene_SRP053398)
)
colnames(mod)
```
Now with the statistical model
```{r}
vGene <- voom(dge, mod, plot = TRUE)
```












